app/models/bookmark.rb

```diff
@@ -0,0 +1,9 @@
+class Bookmark < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :playspace
+
+  validates :user_id, presence: true
+  validates :playspace_id, presence: true
+  validates_uniqueness_of :user_id, scope: [:playspace_id]
+  validates_uniqueness_of :playspace_id, scope: [:user_id]
```

Comment:
I think what you want to say here is that no combination of `user` **and** `playspace` can be repeated for a bookmark, but what you actually said by adding both is that every user can have only one bookmark, and no two users will be able to too bookmark the same playspace :/

http://api.rubyonrails.org/classes/ActiveRecord/Validations/ClassMethods.html#method-i-validates_uniqueness_of